pallets/stake/src/lib.rs

```diff
@@ -42,6 +43,23 @@ pub(crate) mod mock;
 #[cfg(test)]
 mod tests;
 
+#[derive(Copy, Clone, PartialEq, Eq, Encode, Decode, RuntimeDebug)]
+/// Slash status of the validator
+pub enum Slash {
```

Comment:
Why not just keep track of strikes, and 0 strikes means the validator is innocent?

Reply:
Nice, definitely better